benchmark/Main.hs

```diff
@@ -1,2 +1,25 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Main where
+
+import Gauge (bench, bgroup, defaultMain, whnf)
+
+import Time (Day, Nanosecond, Second, ns, sec, toUnit, week)
+
+import qualified Data.Time.Units as TU (Day, Nanosecond, Second, Week, convertUnit)
+
 main :: IO ()
-main = undefined
+main = defaultMain
+    [ bgroup "Second to Nanosecond"
+        [ bench "o'clock"    $ whnf (toUnit @Nanosecond . sec) 1
+        , bench "time-units" $ whnf (TU.convertUnit :: TU.Second -> TU.Nanosecond) 1
+        ]
+    , bgroup "1000ns to s"
+        [ bench "o'clock"    $ whnf (toUnit @Second . ns) 1000
+        , bench "time-units" $ whnf (TU.convertUnit :: TU.Nanosecond -> TU.Second) 1000
+        ]
```

Comment:
Nice! It would be also good to compare with `tiempo` and `data-timeout` packages (of course, only if we can compile them).

Reply:
There is a problem with `Semigroup` instance in `data-timeout`.
Should I try to handle it within the data-timeout library and then add benchs?

Reply:
@vrom911 I think you shouldn't. `data-timeout` library looks not maintained.